src/app/clusters/ota-requestor/DefaultOTARequestor.h

```diff
@@ -205,6 +205,12 @@ class DefaultOTARequestor : public OTARequestorInterface, public BDXDownloader::
      */
     IdleStateReason MapErrorToIdleStateReason(CHIP_ERROR error);
 
+    ScopedNodeId GetProviderScopedId() const
+    {
+        VerifyOrDie(mProviderLocation.HasValue());
```

Comment:
`Value()` already does this check, so it's redundant.

Reply:
Fair enough. It's been a struggle however to get the builds green, and with CI being so slow and this PR blocking a number of other things, I'll merge this as is for now, and see if I can remove this in a later PR.